src/core_plugins/region_map/public/region_map_vis.js

```diff
@@ -8,12 +8,13 @@ import { CATEGORY } from 'ui/vis/vis_category';
 import { VisSchemasProvider } from 'ui/vis/editors/default/schemas';
 import { VisTypesRegistryProvider } from 'ui/registry/vis_types';
 import { truncatedColorMaps } from 'ui/vislib/components/color/truncated_colormaps';
+import { mapToLayerWithId } from './util';
 
 VisTypesRegistryProvider.register(function RegionMapProvider(Private, regionmapsConfig) {
   const VisFactory = Private(VisFactoryProvider);
   const Schemas = Private(VisSchemasProvider);
 
-  const vectorLayers = regionmapsConfig.layers;
+  const vectorLayers = regionmapsConfig.layers.map(mapToLayerWithId.bind(null, 'self_hosted'));
```

Comment:
seems the prefix is always null or 'unknown' ... consider making `prefix` the second parameter, then we can pass it optionally.

Reply:
the prefix can be `elastic_maps_service` or `self_hosted`, which are our two data sources. I removed the `unknown`, as that was an attempt for backward compatibility (saved selectedLayers without ids), something we don't reall yneed for a 6.0 change anyway

Reply:
ehm, looking at the code, second parameter is named layer (self hosted of elastic_maps_service) ... the prefix (first parameter seems unused (you always pass null))

https://github.com/elastic/kibana/pull/14139/files#diff-d6946ea65e6a0b1ff7327c16b279364aR3